### ISSUE TITLE
baresip does not anymore start without network address

### ISSUE THREAD

**Issue by juha-h** [baresip/baresip#1598]
Baresip used to start fine even when there was no other network interfaces than the local one.

Today when I built baresip and tried to start it, I got on Linux:
```
$ baresip
baresip v1.1.0 Copyright (C) 2010 - 2021 Alfred E. Heggestad et al.
net: : could not get network address
ua: network init failed: Cannot assign requested address
main: baresip init failed (Cannot assign requested address)
ua: stop all (forced=1)
```
and on Android:
```
09-16 08:29:02.368  1532  1597 D Baresip Lib: starting baresip
09-16 08:29:02.376  1532  1598 D Baresip Lib: net: : could not get network address
09-16 08:29:02.376  1532  1597 D Baresip Lib: baresip_init() failed (99)
09-16 08:29:02.377  1532  1597 D Baresip Lib: stopping UAs due to error: (99)
09-16 08:29:02.377  1532  1597 D Baresip Lib: closing ...
```

**Comment by juha-h** [comment]
Here is what I get using baresip Android built in August:
```
09-16 08:41:01.145  1798  1798 D Baresip Lib: starting baresip
09-16 08:41:01.164  1798  1834 D Baresip Lib: Local network address:  
09-16 08:41:01.165  1798  1833 D Baresip Lib: --- Network debug ---
09-16 08:41:01.165  1798  1833 D Baresip Lib:  Local IPv4:  [E] (not set)
09-16 08:41:01.165  1798  1833 D Baresip Lib:  Local IPv6:  [E] (not set)
09-16 08:41:01.165  1798  1833 D Baresip Lib: net interfaces:
09-16 08:41:01.165  1798  1833 D Baresip Lib:          lo:  127.0.0.1
09-16 08:41:01.165  1798  1833 D Baresip Lib:  DNS Servers from System: (0)
```

**Comment by juha-h** [comment]
A verified this by building libbaresip from re and baresip masters as whey were on  on Aug 25, and then baresip started fine w/o network address.  I then built libbaresip from today's re and baresip masters, and got the above described start failure.

**Comment by webstean** [comment]
This is the correct behaviour? Right - With no IP address what is the point of starting baresip?

**Comment by juha-h** [comment]
Sure baresip must start like it has been doing for several years.  It is common that, for example, Android user starts baresip and then realizes that he/she has forgotten to enable wifi or mobile data.  It must be enough that network is enabled without a need to start baresip again.

**Comment by alfredh** [comment]
the most flexible approach is that baresip can start without internet access.
for example, the network can come up 1 second after the app started.

**Comment by webstean** [comment]
Sounds like baresip is trying to solve a problem that the OS/user should really solve ( ie don't start baresip until there is a network ).
Accept that baresip has been doing this itself for years, but if baresip is to be as 'bare' as possible then l'd question why it needs the additional logic. lm not an Android users, so l have no idea about that platform but l would agree that having baresip keep working when the ip address changes for a mobile user would be a good thing which l believe is what it does today.

Perhaps a init-key-trying option might be the answer here, if net init fails then keep trying if this option is set otherwise exit and tell the user to ensure dependencies like network is already running.

**Comment by juha-h** [comment]
Polling is bad idea due to CPU usage and thus battery consumption.  Everything should be based on events.  baresip should start without network access and should not stop running if network access is lost.  Currently I use net_add|rm_address to tell baresip about changes in network connectivity.

**Comment by webstean** [comment]
Who said anything polling, why not just sleep if init fals and try again in 30 seconds.

**Comment by cspiel1** [comment]
Both is a senseful wanted behavior.
But because the previous behavior was that starting without network was possible, and because the functionality of network update (manual and automatic) is already implemented, we should prefer this like @alfredh and @juha-h wish. I will have a look on this today.

**Comment by sreimers** [comment]
Fixed by #1607
